### PR TITLE
feat: refactor management/Makefile to make upgrade easier. Part of #90 and #158

### PR DESCRIPTION
**Pull request by Bobgy**
Part of https://github.com/kubeflow/gcp-blueprints/issues/90 and https://github.com/kubeflow/gcp-blueprints/issues/158.

Changes:
1. make future upgrading management package as simple as `make update`, even there are makefile changes
2. better separation of concerns
    * add a `packages/management/Makefile` that can be used to deploy the management package directly
    * but it can also be configured as a recursive Makefile in an upstream package
3. update docs to reflect current status
4. add a `set-values.sh` script where users can set their values

All of the above changes are completely compatible with existing `instance` folder setup, so users won't need to touch `instance` folder during upgrade.

### CODE REVIEW

**Comment by Bobgy** [comment]
/assign @chensun @jlewi @subodh101 
for review

File: management/Makefile [hunk 1]
```diff
@@ -1,132 +1,76 @@
...
-# Directory where manifests should be fetched to
-MANIFESTS_DIR=./upstream/management
+PACKAGE_URL?=https://github.com/kubeflow/gcp-blueprints.git/packages/management@master
 
 INSTANCE_DIR=./instance
 
+# Directory where packages should be read from. Export these variables, so
+# recursive make commands can get them.
+export PACKAGE_DIR=./upstream/management
+export PACKAGE_CLUSTER_DIR=$(INSTANCE_DIR)/cluster
```
Comment (chensun) [inline management/Makefile:18]: For my education, why do we have a variable for instance folder but not for `./upstream` for instance? Will there be multi-instance support?

File: management/set-values.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#
+# The values you set will be stored in ./instance/Kptfile and
+# ./upstream/management/Kptfile, so they will be preserved if you commit the
+# changes to source control.
+
+set -ex
+
+for package in (./instance ./upstream/management)
+do
+    kpt cfg set $package name $NAME
```
Comment (chensun) [inline management/set-values.sh:19]: nit: prefer quoting variables (ref: https://google.github.io/styleguide/shellguide.html#variable-expansion)

**Review by chensun — COMMENTED** [review]
/lgtm

File: management/Makefile (hunk 1 above)
Comment (Bobgy) [reply management/Makefile:18]: Because the instance folder is just an empty kustomization.yaml that points to the upstream folder.

**Comment by Bobgy** [comment]
Hmmm, the new auto approval is kind of confusing.

Feel free to add more commnents and I'll address them in a separate PR.

File: management/set-values.sh (hunk 2 above)
Comment (Bobgy) [reply management/set-values.sh:19]: Thanks, will do

**Pull request merged**